PowerEditor/src/NppIO.cpp

```diff
@@ -1328,8 +1328,9 @@ bool Notepad_plus::fileSaveAs(BufferID id, bool isSaveCopy)
 
 	if (pfn)
 	{
+		BufferID same = _pDocTab->findBufferByName(pfn);

```

Comment:
It seems second search is not really needed if first search is successful. So I think it's enough to have just one variable (maybe renamed to sameFile) and just do second search only if first search is failed.